### PR TITLE
feat(summarization): persist sourceEvidence and depthLift on suggestion

### PR DESCRIPTION
**Pull request by anuj-adobe**
## Summary

Persists two new fields onto suggestion records in the DB via \`getJsonSummarySuggestion\` in \`src/summarization/utils.js\`:

- **\`sourceEvidence\`** — verbatim raw-page sentences captured by the gen_summary_crew that ground each key point / summary fact. Used by the on-demand prompt-regeneration path (\`build_summarization_inputs\`) as \`kp_sources\` / \`summary_sources\`, closing the gap where re-generation had no evidence anchoring (mystique PR #3395).
- **\`depthLift\`** — page-level signal (0..1) measuring how deeply buried the facts behind a page's prompts are in the page content. Used for High Impact URL ranking: \`High Impact = agentic_traffic × depth_lift\`. Stored on both suggestion records (summary + key-points) so the UI can use it without extra fetching.

Both fields default safely when absent: \`sourceEvidence → []\`, \`depthLift → 0\`. Backward compatible with existing suggestions and the existing-summary/key-points path.

## Test plan

- [ ] \`npm run test:spec -- test/audits/summarization/utils.test.js\` — 28 tests covering both fields (present, absent, invalid-type cases)
- [ ] Full \`npm test\` passes
- [ ] Verify suggestions in DB have \`sourceEvidence\` and \`depthLift\` populated after a summarization audit run

## Related

- mystique PR: https://git.corp.adobe.com/experience-platform/mystique/pull/3553 — computes \`depth_lift\` signal, captures verbatim \`source_evidence\` per key point and summary fact

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/summarization/utils.js [hunk 1]
```diff
@@ -59,6 +59,10 @@ export function getJsonSummarySuggestion(suggestions, urlToContentHash = {}) {
     const keyPointsPrompts = Array.isArray(suggestion.keyPointsPrompts)
       ? suggestion.keyPointsPrompts : [];
 
+    // depthLift is a page-level signal (mean burial weight of facts behind prompts, 0..1).
```
Comment (MysticatBot) [inline src/summarization/utils.js:62]: issue (blocking): `typeof suggestion.depthLift === 'number'` passes `NaN` and `Infinity`/`-Infinity` since `typeof NaN === 'number'` evaluates to `true`. These values serialize as `null` in JSON (via `JSON.stringify`), so the stored record would get `null` rather than the intended default of `0`. The UI consuming `depthLift` for High Impact URL ranking would then receive `null` instead of a valid numeric signal.

Fix: replace with `Number.isFinite(suggestion.depthLift)`:
```js
const depthLift = Number.isFinite(suggestion.depthLift) ? suggestion.depthLift : 0;
```
This correctly rejects `NaN`, `Infinity`, `-Infinity`, and non-number types in a single guard.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @anuj-adobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - one edge case in the depthLift guard.
**Complexity: MEDIUM** - medium diff, mostly tests.
_Changes: Persists `depthLift` and `sourceEvidence` onto summary/key-points suggestion records for downstream prompt regeneration (2 files)._
Note: CI checks are currently failing - resolve before merge.

## Must fix before merge
1. [Important] `depthLift` guard passes NaN/Infinity through - `src/summarization/utils.js:62` (details inline)

<details>
<summary>Non-blocking (2): minor issues and suggestions</summary>

- suggestion: Add a size cap on `sourceEvidence` arrays before persistence (e.g. `.slice(0, 50)`) to guard against unbounded upstream payloads being fed into token-limited prompts downstream - `src/summarization/utils.js:84`
- suggestion: Validate that `sourceEvidence` elements are strings (`.filter(e => typeof e === 'string')`) to prevent non-string values from reaching downstream prompt construction - `src/summarization/utils.js:84`

</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 2m 54s | <b>Cost:</b> $3.63 | <b>Commit:</b> <code>67a058264bffc68e9c89f6c22867306a5d8e2a88</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @anuj-adobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - all prior findings addressed, code is clean and well-tested.
**Complexity: MEDIUM** - medium diff, mostly tests.
_Changes: Persists `depthLift` and `sourceEvidence` onto summary/key-points suggestion records for downstream prompt regeneration and High Impact URL ranking (2 files)._

## Previously flagged, now resolved
- `depthLift` NaN/Infinity guard now uses `Number.isFinite()` with fallback to `0`
- `sourceEvidence` capped at 50 entries via `.slice(0, 50)`
- Non-string elements filtered via `.filter((e) => typeof e === 'string')`

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 0m 59s | <b>Cost:</b> $2.05 | <b>Commit:</b> <code>e2fc3fce3e3a32ee8cfc419460a64f891c71e6d5</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by Danialkafeel — APPROVED** [review]
